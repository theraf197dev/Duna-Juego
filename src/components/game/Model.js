import React, { Component }  from "react";
import {
    CellType,
    GameMode,
    GameStatus,
} from '../../common/utils';
import {setInitBoard} from "./helper";
import Controller from "./Controller";
import seekerMove from "./helper/seeker";

export default class Model extends Component {
    constructor(props) {
        super();
        this.state = {
            coverSeekerValue: CellType.pizza,
            currentPos : {
                x: Math.floor((props.size - 1) / 2),
                y: Math.floor((props.size - 1) / 2),
            },
            gameStatus: GameStatus.playing,
            lastInput: {
                incr: 1,
                vector: 'y',
            },
            matrix: Array.from({length: props.size}, ()=> Array.from({length: props.size}, () => CellType.blank)),
            pizzas: 1,
            seekerCurrentPos: {
                x: 0,
                y: 0,
            },
            seekerLastMovement: {
                incr: 1,
                vector: 'x',
            },
        };
        this.bindActions();
    };

    bindActions() {
        this.setCurrentPos = this.setCurrentPos.bind(this);
        this.setMatrix = this.setMatrix.bind(this);
        this.stopInterval = this.stopInterval.bind(this);
        this.endGame = this.endGame.bind(this);
    }

    componentDidMount() {
        this.setInitialMatrix();
        if(this.props.mode === GameMode.seeker){
            this.interval = setInterval(() => {
                const {vector, incr} = seekerMove(
                    this.state.seekerLastMovement,
                    this.state.seekerCurrentPos,
                    this.state.matrix,
                    this.props.size
                );
                this.setSeekerPos(vector, incr);
            }, 500);
        }
    }

    componentWillUnmount() {
        if(this.props.mode === GameMode.seeker){
            this.stopInterval();
        }
    }

    stopInterval() {
        clearInterval(this.interval);
    }

    setPizzas(pizzas) {
        this.setState({
            pizzas: pizzas,
        });
    };

    endGame(status) {
        this.setState({
            gameStatus: status,
        });
    };

    setCurrentPos(vector, incr) {
        const nextPos = {
            ...this.state.currentPos,
            [vector]: this.state.currentPos[vector] + incr,
        };

        this.setState({
            currentPos: nextPos,
            matrix: this.state.matrix.map((col, x) =>
                col.map((cell) => cell !== CellType.duna ? cell : CellType.blank)
            ),
            lastInput: {
                incr,
                vector,
            },
            pizzas: this.state.matrix[nextPos.x][nextPos.y] === CellType.pizza ?
                this.state.pizzas - 1 : this.state.pizzas,
        });

        if(this.state.matrix[nextPos.x][nextPos.y] === CellType.seeker) {
            this.endGame(GameStatus.defeat);
        }
    };

    setSeekerPos(vector, incr) {
        const nextSeekerPos = {
            ...this.state.seekerCurrentPos,
            [vector]: this.state.seekerCurrentPos[vector] + incr,
        };

        this.setState({
            seekerCurrentPos: {
                ...this.state.seekerCurrentPos,
                [vector]: this.state.seekerCurrentPos[vector] + incr,
            },
            seekerLastMovement: {
                incr,
                vector,
            },
            matrix: this.state.matrix.map((col, x) =>
                col.map((cell) => cell !== CellType.seeker ? cell : this.state.coverSeekerValue)
            ),
            coverSeekerValue: this.state.matrix[nextSeekerPos.x][nextSeekerPos.y],
        });

        if(this.state.matrix[nextSeekerPos.x][nextSeekerPos.y] === CellType.duna) {
            this.endGame(GameStatus.defeat);
        }
    };

    setMatrix(value, pos = this.state.currentPos) {
        const {x, y} = pos;
        this.setState({
            matrix: this.state.matrix.map((col, index) =>
                index === x ?
                col.map((cell, index) => index === y ? value : cell)
                : col
            )
        });
    };

    setInitialMatrix() {
        const { board, pizzas } = setInitBoard(this.props.size, this.props.mode, this.props.difficulty);
        this.setState({
            matrix: this.state.matrix.map((col, x) =>
                col.map((cell, y) => board[x][y])
            ),
            pizzas: pizzas,
        });
    };

    render() {
        return (
            <Controller
                {...this.props}
                {...this.state}
                endGame={this.endGame}
                setCurrentPos={this.setCurrentPos}
                setMatrix={this.setMatrix}
                stopInterval={this.stopInterval}
            />
        )
    };
};

Model.displayName = 'ModelGame';

