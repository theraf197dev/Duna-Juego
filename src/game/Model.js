import React, { Component }  from "react";
import {
    CellType,
} from '../common/utils';
import setInitBoard from "./helper";
import Controller from "./Controller";

export default class Model extends Component {
    constructor(props) {
        super();
        this.state = {
            matrix: Array.from({length: props.size}, ()=> Array.from({length: props.size}, () => CellType.blank)),
            currentPos : {
                x: Math.floor((props.size - 1) / 2),
                y: Math.floor((props.size - 1) / 2),
            },
            seekerCurrentPos: {
                x: 0,
                y: 0,
            },
            coverSeekerValue: CellType.pizza,
            timer: 0,
            isGameAlive: true,
            lastInput: {
                vector: 'y',
                incr: 1,
            },
        };
        this.bindActions();
    };

    bindActions() {
        this.setCurrentPos = this.setCurrentPos.bind(this);
        this.setMatrix = this.setMatrix.bind(this);
        this.setLastInput = this.setLastInput.bind(this);
        this.stopInterval = this.stopInterval.bind(this);
        this.endGame = this.endGame.bind(this);
    }

    componentDidMount() {
        this.setInitialMatrix();
        if(this.props.mode === 'seeker'){
            this.interval = setInterval(() => {
                this.setSeekerPos('x', 1);
            }, 3000);
        }
    }

    componentWillUnmount() {
        if(this.props.mode === 'seeker'){
            this.stopInterval();
        }
    }

    stopInterval() {
        clearInterval(this.interval);
    }

    endGame() {
        this.props.setGame(false);
    }

    setCurrentPos(vector, incr) {
        this.setState({
            currentPos: {
                ...this.state.currentPos,
                [vector]: this.state.currentPos[vector] + incr,
            },
            matrix: this.state.matrix.map((col, x) =>
                col.map((cell) => cell !== CellType.duna ? cell : CellType.blank)
            ),
        });
    };

    setSeekerPos(vector, incr) {
        const nextSeekerPos = {
            ...this.state.seekerCurrentPos,
            [vector]: this.state.seekerCurrentPos[vector] + incr,
        };

        this.setState({
            seekerCurrentPos: {
                ...this.state.seekerCurrentPos,
                [vector]: this.state.seekerCurrentPos[vector] + incr,
            },
            matrix: this.state.matrix.map((col, x) =>
                col.map((cell) => cell !== CellType.seeker ? cell : this.state.coverSeekerValue)
            ),
            coverSeekerValue: this.state.matrix[nextSeekerPos.x][nextSeekerPos.y],
        });
    };

    setMatrix(value, pos = this.state.currentPos) {
        const {x, y} = pos;
        this.setState({
            matrix: this.state.matrix.map((col, index) =>
                index === x ?
                col.map((cell, index) => index === y ? value : cell)
                : col
            )
        });
    };

    setInitialMatrix() {
        const board = setInitBoard(this.props.size, this.props.mode, this.props.difficulty);
        this.setState({
            matrix: this.state.matrix.map((col, x) =>
                col.map((cell, y) => board[x][y])
            )
        });
    };

    setLastInput(value) {
        this.setState({
            lastInput: value,
        });
    };

    render() {
        return (
            <Controller
                {...this.props}
                {...this.state}
                endGame={this.endGame}
                setCurrentPos={this.setCurrentPos}
                setLastInput={this.setLastInput}
                setMatrix={this.setMatrix}
                stopInterval={this.stopInterval}
            />
        )
    };
}

