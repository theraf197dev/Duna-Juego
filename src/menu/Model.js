import React, { Component }  from "react";
import Controller from "./Controller";

export default class Model extends Component {
    constructor(props) {
        super(props);
        this.state = {
            difficulty: props.difficulty,
            isGameAlive: false,
            mode: props.mode,
        };
        this.bindActions();
    };

    bindActions() {
        this.setGame = this.setGame.bind(this);
        this.startGame = this.startGame.bind(this);
    }

    startGame(difficulty, mode) {
        this.setState({
            difficulty,
            isGameAlive: true,
            mode,
        });
    }

    setGame(value) {
        this.setState({
            isGameAlive: value,
        });
    }

    render() {
        return (
            <Controller
                {...this.props}
                {...this.state}
                setGame={this.setGame}
                startGame={this.startGame}
            />
        )
    };
}

Model.displayName = 'ModelMenu';
