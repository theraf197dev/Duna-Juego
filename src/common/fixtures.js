import trialImage from './media/trial_mode.jpg';
import seekerImage from './media/seeker_mode.jpg';
import { CellType, GameMode } from './utils';

const controls = [
    {
        id: 1,
        name: 'Up',
        vector: 'x',
        incr: -1,
    },
    {
        id: 2,
        name: 'Down',
        vector: 'x',
        incr: 1,
    },
    {
        id: 3,
        name: 'Left',
        vector: 'y',
        incr: -1,
    },
    {
        id: 4,
        name: 'Right',
        vector: 'y',
        incr: 1,
    },
];

const games = {
    items: [
        {
            name: GameMode.trial,
            url: trialImage,
        },
        {
            name: GameMode.seeker,
            url: seekerImage,
        },
    ],
};

const matrix = [
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.duna, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
    [CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank, CellType.blank],
];

const difficulty = 1;
const mode = GameMode.trial;
const size = 9;
const time = 30;

const fixtures =  {
    controls,
    difficulty,
    games,
    matrix,
    mode,
    size,
    time,
};

export default fixtures;
