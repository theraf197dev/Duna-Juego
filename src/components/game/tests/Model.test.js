import { render } from "@testing-library/react";
import fixtures from '../../../common/fixtures';
import Model from "../Model";

describe('Model tests', () => {
  let props = {
    ...fixtures,
    restartGame: jest.fn(),
  };

  it('render component', () => {
      const { container } = render(<Model {...props} />);

      expect(container).toMatchSnapshot();
  });
});
