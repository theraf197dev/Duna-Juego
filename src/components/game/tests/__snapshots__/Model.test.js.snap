// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model tests render component 1`] = `
<div>
  <div
    class="sc-bcXHqe gkqtbX"
  >
    <div
      class="sc-ksBlkl jmQgxj"
    >
      00 : 30
    </div>
    <div
      class="sc-gswNZR cDelEv"
    >
      <div
        class="sc-dkrFOg hUHXsd"
      >
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe bnTpRO"
              src="duna.jpg"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
        </tr>
        <tr>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="blank.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
          <td
            class="sc-hLBbgP hIhZhQ"
          >
            <img
              class="sc-eDvSVe epBgfN"
              src="pizza.png"
            />
          </td>
        </tr>
      </div>
      <div
        class="sc-jSUZER gOoPkq"
      >
        <button
          class="sc-gKPRtg hmnFoD"
        >
          <div
            class="sc-iBYQkv jMtJtr"
          />
        </button>
        <button
          class="sc-gKPRtg cILazt"
        >
          <div
            class="sc-iBYQkv icdsrj"
          />
        </button>
        <button
          class="sc-gKPRtg hXaRCV"
        >
          <div
            class="sc-iBYQkv RaszP"
          />
        </button>
        <button
          class="sc-gKPRtg hoHrNM"
        >
          <div
            class="sc-iBYQkv ALwnz"
          />
        </button>
        <div
          class="sc-pyfCe droXxt"
        >
          <div
            class="sc-ftTHYK eFTIwb"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
